ca/certificates.go

```diff
@@ -415,6 +390,100 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 	return RootCA{Signer: signer, Key: keyBytes, Digest: digest, Cert: certBytes, Pool: pool}, nil
 }
 
+// ValidateCertChain checks that that a bundle certificates for a chain, each one certifying the one above,
+// as per RFC5246 section 7.4.2. They must also form at least one complete chain up to the root pool provided.
+// If everything validates, the entire parsed chain is returned.  Note that the chain can contain some unused
+// certificates (ones that aren't used to form a chain up to the given root pool).  This validation also enforces
+// that none of the certificates in the chain can be expired (unless allowExpired is set to true) or not yet
+// valid. This always returns at least 1 certificate.
+func ValidateCertChain(rootPool *x509.CertPool, certs []byte, allowExpired bool) ([]*x509.Certificate, error) {
```

Comment:
We wanted to be able to load expired certs when forcing a new cluster, because then we are the signer and can produce new certs.  Otherwise you can't really recover a cluster easily using force new cluster.